web/src/utils.jsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+if (!process.env.ABSOLUTE_LINK_PREFIX) {
+  throw new Error("ABSOLUTE_LINK_PREFIX is not set");
+}
+
+const absoluteLinkPrefix = process.env.ABSOLUTE_LINK_PREFIX;
+
+export function toAbsoluteIncomingLink(relativeIncomingLink) {
+  const link = relativeIncomingLink
+    ? absoluteLinkPrefix + relativeIncomingLink
+    : "";
+  return toLinkOpeningNewTab(link);
+}
+
+export function toLinkOpeningNewTab(link) {
+  return (
+    <a href={link} target="_blank" rel="noopener noreferrer">
+      {link}
+    </a>
+  );
+}
```

Comment:
There are two coupling problems here, in my opinion.

First, there are 2 independent functionalities here: expanding a relative incoming URL to an absolute one, and forming an JSX link from an URL. Despite those functionalities being independent, in the current implementation, it is impossible to use the first without the second, because the it *happens* (by chance) that the rest of the program never uses the first one without the second one, so it is *slightly* more convenient to call only function instead of two. This means that while this is a module, which is supposed to be independent, it is actually coupled to the rest of the program, strongly. The solution is for `toAbsoluteIncomingLink` to simply do its job and return the absolute URL.

Second, naming this file `utils.jsx` encourages future contributors to add unrelated stuff into this file. Actually this is already the case. These are two *unrelated* functionalities. Also, more concretely, putting them in the same file makes it impossible to use `toAbsoluteIncomingLink` without importing React into the app. Solution : put each function into its own file.

What do you think?